src/NumericalAlgorithms/RootFinding/NewtonRaphson.hpp

```diff
@@ -1,120 +1,18 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
-/// \file
-/// Declares function RootFinder::newton_raphson
-
 #pragma once
 
-#include <boost/math/tools/roots.hpp>
-#include <functional>
-#include <limits>
-
-#include "DataStructures/DataVector.hpp"
-#include "Utilities/ErrorHandling/Exceptions.hpp"
-#include "Utilities/MakeString.hpp"
+#error "RootFinding/NewtonRaphson.hpp does not provide any functionality."
 
 namespace RootFinder {
 /*!
  * \ingroup NumericalAlgorithmsGroup
- * \brief Finds the root of the function `f` with the Newton-Raphson method.
- *
- * `f` is a unary invokable that takes a `double` which is the current value at
- * which to evaluate `f`. `f` must return a `std::pair<double, double>` where
- * the first element is the function value and the second element is the
- * derivative of the function.  An example is below.
- *
- * \snippet Test_NewtonRaphson.cpp double_newton_raphson_root_find
- *
- * See the [Boost](http://www.boost.org/) documentation for more details.
- *
- * \requires Function `f` is invokable with a `double`
- * \note The parameter `digits` specifies the precision of the result in its
- * desired number of base-10 digits.
- *
- * \throws `convergence_error` if the requested precision is not met after
- * `max_iterations` iterations.
+ * For the nonlinear solver, see NonlinearSolver::newton_raphson.  We
+ * do not provide a Newton-Raphson root finder.  The Boost
+ * implementation is buggy and can return the wrong answer, and it is
+ * not clear that performance would be better than RootFinder::toms748
+ * in practice.
```

Comment:
Maybe add a sentence that if the derivatives are cheap to evaluate NR is generally better, but for practical cases where the function and the derivative are the dominant cost of the root find, TOMS748 is comparable? There's some discussion on the Boost website that makes it look like NR is better, but in the caveats they explicitly state this is just because of overhead, not root finding speed per-se.

Reply:
Yes, and then advertise it 🙃 I think the new paragraph is very helpful! Thanks for adding that!